com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/validators/v3/ExtensionDescriptorValidator.java

```diff
@@ -41,16 +41,16 @@ protected void validateProperties(PropertiesContainer container, PropertiesConta
 
     protected void validate(Metadata metadata, Map<String, Object> properties, Map<String, Object> extensionProperties,
         String containerName, String elementType) {
-        for (String propertyName : extensionProperties.keySet()) {
-            boolean isOverwritable = metadata.getOverwritableMetadata(propertyName);
-            Object value = properties.get(propertyName);
-            Object extensionValue = extensionProperties.get(propertyName);
-            if (!isOverwritable && extensionValue != null) {
+        for (Map.Entry<String, Object> property : extensionProperties.entrySet()) {
```

Comment:
It is now hard to distinguish between `property.getValue()` and `value`. Rename the `property` variable to `extensionProperty`.